scribus/text/storytext.cpp

```diff
@@ -1672,24 +1673,14 @@ int StoryText::selectWord(int pos)
 {
 	//Double click in a frame to select a word
 
-	int a = pos;
-	while(a > 0)
-	{
-		if (text(a-1).isLetterOrNumber())
-			--a;
-		else
-			break;
-	}
-	int b = pos;
-	while(b < length())
-	{
-		if (text(b).isLetterOrNumber())
-			++b;
-		else
-			break;
-	}
-	select(a, b - a);
-	return a;
+	UErrorCode status = U_ZERO_ERROR;
+	BreakIterator* bi = BreakIterator::createWordInstance(Locale::getUS(), status);
```

Comment:
Indeed QTextBoundaryFinder & BreakIterator are slow but in other side, It will used once or twice which makes caching difficult nor  give a big difference in performance in the end.